src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,117 @@
+import {Dimensions} from 'react-native';
+import spacing from './utilities/spacing';
+import styles from './styles';
+import themeColors from './themes/default';
+import fontFamily from './fontFamily';
+import variables from './variables';
+
+const GUTTER_WIDTH = 16;
+
+/**
+ * The Expensify.cash repo is very consistent about doing spacing in multiples of 4.
+ * In an effort to maintain that consistency, we'll make sure that any distance we're shifting the tooltip
+ * is a multiple of 4.
+ *
+ * @param {Number} n
+ * @returns {Number}
+ */
+function roundToNearestMultipleOfFour(n) {
+    if (n > 0) {
+        return Math.ceil(n / 4.0) * 4;
+    }
+
+    if (n < 0) {
+        return Math.floor(n / 4.0) * 4;
+    }
+
+    return 0;
+}
+
+export default function getTooltipStyles(
+    interpolatedSize,
+    xOffset,
+    yOffset,
+    width,
+    height,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    const windowWidth = Dimensions.get('window').width;
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    let horizontalShift = 0;
+    const tooltipLeftEdge = (xOffset + (width / 2)) - (tooltipWidth / 2);
+    const tooltipRightEdge = (xOffset + (width / 2)) + (tooltipWidth / 2);
```

Comment:
So `xOffset` is always measured from the left edge.... and we are dividing the width of _something_ by `2` (why?) and then also dividing the tooltip width by `2`. Having a hard time visualizing this maybe because I'm unsure what `width` is referring to.

Reply:
Renamed `width` -> `componentWidth`. Not sure how useful such a comment would be, but what we're doing is:

1. `left edge of the wrapped component` + `half the component's width` = `center of component`
2. `center of component` - `half the tooltip's width` = `left edge of tooltip` (because the tooltip is centered on the component by default.

Reply:
I would say that is super useful. I'm kind of a visual learner and guessing others are as well so painting a bit of a picture in words seems like a good idea for some of these.

Reply:
Moved the horizontal shifting logic to its own function and I think it's easier to understand now.